core/src/test/java/org/elasticsearch/test/geo/RandomShapeGenerator.java

```diff
@@ -211,7 +219,8 @@ private static ShapeBuilder createShape(Random r, Point nearPoint, Rectangle wit
                     try {
                         pgb.build();
                     } catch (InvalidShapeException e) {
-                        return null;
+                        // jts bug rarely results in an invalid shape, if it does happen we try again instead of returning null
+                        return createShape(r, nearPoint, within, st, validate);
```

Comment:
Not at all common. Typically (~90% of the time) the first recursion creates something other than a polygon (e.g., linestring, multilinestring) thus completes without issue.